src/librustc_parse/parser/mod.rs

```diff
@@ -662,6 +662,24 @@ impl<'a> Parser<'a> {
                                 }
                             }
 
+                            // If this was a missing `@` in a binding pattern
+                            // bail with a suggestion
+                            // https://github.com/rust-lang/rust/issues/72373
+                            if self.prev_token.is_ident() && &self.token.kind == &token::DotDot {
+                                let msg = &format!("if you meant to bind the contents of \
+                                    the rest of the array pattern into `{}`, use `@`",
+                                    pprust::token_to_string(&self.prev_token));
+                                expect_err
+                                    .span_suggestion_verbose(
+                                        self.token.span,
```

Comment:
To always have consistent space between the ident and `@`, do this:
```suggestion
                                        self.prev_token.span.shrink_to_hi().until(self.token.span),
```

Reply:
Sorry, didn't come across this comment yet.  I'll do this. Ty!

Reply:
with the change of the suggested string to ` @ ` this works irrespective of the original spacing between the ident and `@`:

```
error: expected one of `,`, `@`, `]`, or `|`, found `..`
 --> src/main.rs:5:19
  |
5 |         [h, ref ts..] => foo(c, n - h) + foo(ts, n),
  |                   ^^ expected one of `,`, `@`, `]`, or `|`
  |
help: if you meant to bind the contents of the rest of the array pattern into `ts`, use `@`
  |
5 |         [h, ref ts @ ..] => foo(c, n - h) + foo(ts, n),
  |                    ^

error: aborting due to previous error
```

```
error: expected one of `,`, `@`, `]`, or `|`, found `..`
 --> src/main.rs:5:20
  |
5 |         [h, ref ts ..] => foo(c, n - h) + foo(ts, n),
  |                    ^^ expected one of `,`, `@`, `]`, or `|`
  |
help: if you meant to bind the contents of the rest of the array pattern into `ts`, use `@`
  |
5 |         [h, ref ts @ ..] => foo(c, n - h) + foo(ts, n),
  |                    ^

error: aborting due to previous error
```